### ISSUE TITLE
UIP-3197 Ignore imports in strings

### ISSUE THREAD

**Issue by nickclmb** [Workiva/dependency_validator#21]
At the moment dependency validator thinks that this is real import:
```dart
void main() {
  final content = '''import 'package:test/test.dart';'''
  ...
}
```

**Comment by evanweible-wf** [comment]
Thanks for reporting this @nickclmb! We currently use a regex to quickly detect imports in files, which is why this is getting picked up. This one might be tough to fix without properly using the analyzer to build the AST... any ideas @Workiva/app-frameworks?

**Comment by aaronlademann-wf** [comment]
@evanweible-wf a "real" import would always start with the word `import`, so we could use the `^` character in the regex to specify that?

**Comment by nickclmb** [comment]
Hi guys, the `^` character will undo the feature when multiple imports written in one row.
Probably something like this will work:
```dart
final RegExp importExportPackageRegex = 
     new RegExp(r'''^(\s*(import|export)\s+['"]{1,3}package:([a-zA-Z0-9_]+)\/[^;]+['"]{1,3};)+'''); 
``` 
it wraps single import into group, handles multiple imports in a row, and ignore strings like
```dart
'import "package:name/path.dart"';
// import 'package:name/path.dart';
```

**Comment by aaronlademann-wf** [comment]
@evanweible-wf ^

**Comment by jacehensley-wf** [comment]
I think Evan is right, the real path forward would be to use the AST

**Comment by greglittlefield-wf** [comment]
@evanweible-wf Agreed; we could make adjustments to the Regex to fix this case, but I think our time would be better spent just implementing the AST solution (which I don't foresee taking much time) vs wrestling with edge cases and the Regex.

**Comment by matthewnitschke-wk** [comment]
This issue has been resolved in #111, and will be included in the `v4.0.0` release
